docs/specs/tsd-008-partial-write-error-handling.md

```diff
@@ -0,0 +1,59 @@
+# Partial write error handling
+
+## Objective
+
+Provide a way to pass information about partial metric write errors from an
+output to the output model.
+
+## Keywords
+
+output plugins, write, error, output model, metric, buffer
+
+## Overview
+
+When output plugins serialize metric and/or send them to the service endpoint,
+single metrics might cause errors e.g. by not being serializable or by being
+rejected by the service on the server side.
+Currently, an output is only able to accept or reject the complete batch of
+metrics it receives from the output model. This causes issues if only a subset
+of metrics in the batch fails as the output has no way of telling the model
+(and in turn the output buffer) which metrics failed but can only accept or
+reject the whole batch. As a consequence, outputs need to "accept" the batch
+to avoid a requeueing of the batch for the next flush interval. This distorts
+statistics of accepted metrics and causes misleading log messages.
```

Comment:
Can you give an example? Again I'm thinking of InfluxDB where an error is thrown, but we also say things are accepted.

Reply:
The example is issue #11942 where you see `wrote all metrics successfully` but half of them are known to be dropped by the InfluxDB server. Even worse, if you are looking into the statistics reported by the `inputs.internal` plugin you won't see _any_ dropped metrics at all.